Simplified/AxisService/NYPLRSACypher.swift

```diff
@@ -126,52 +128,95 @@ struct NYPLRSACypher: NYPLRSACryptographing {
     
     self.publicKey = publicKey.base64EncodedString()
     self.privateKey = privateKey.base64EncodedString()
-    self.modulus = self.publicKey.replacingOccurrences(of: "/", with: "-")
-    // we're creating a key with 2048 bits. Exponent for that is AQAB.
-    self.exponent = "AQAB"
-  }
-  
-  func encryptText(_ message: String) -> String? {
-    let _messageData = message.data(using: .utf8)
     
-    guard let messageData = _messageData else {
+    guard
+      let (mod, exp) = NYPLRSACypher.parsePublicSecKey(publicKey: self.publicKeySec)
+    else {
       NYPLRSACypher
-        .logFailedEncryptionError(NYPLRSACypher.stringToDataConversionFailure)
-      return nil
-    }
-    guard let encryptedData = SecKeyCreateEncryptedData(publicKeySec,
-                                                        .rsaEncryptionOAEPSHA1,
-                                                        messageData as CFData,
-                                                        nil)
-      else {
-        NYPLRSACypher
-          .logFailedEncryptionError(NYPLRSACypher.nilEncryptedDataFailure)
+        .logFailedInitializationError(
+          "Failed to get exponent and or modulus for public key")
         return nil
     }
     
-    return (encryptedData as Data).base64EncodedString()
+    let modulus = mod.base64EncodedString()
+    guard modulus.count >= 342 else {
+      NYPLRSACypher
+        .logFailedInitializationError("Got incorrect modulus from public key")
+      return nil
+    }
+    
+    self.modulus = String(modulus.prefix(342)).replacingOccurrences(of: "/", with: "_")
+    self.exponent = exp.base64EncodedString()
   }
   
-  func decryptText(_ encryptedString: String) -> String? {
-    let _messageData = Data(base64Encoded: encryptedString)
+  static private func parsePublicSecKey(publicKey: SecKey) -> (mod: Data, exp: Data)? {
     
-    guard let messageData = _messageData else {
-      NYPLRSACypher
-        .logFailedDecryptionError(NYPLRSACypher.stringToDataConversionFailure)
+    guard let pubAttributes = SecKeyCopyAttributes(publicKey) as? [String: Any] else {
       return nil
     }
     
+    let keySize = pubAttributes[kSecAttrKeySizeInBits as String] as! Int
+    let pubData = pubAttributes[kSecValueData as String] as! Data
+    var modulus = pubData.subdata(in: 8..<(pubData.count - 5))
```

Comment:
I don't know what's going on with these magic numbers...

or why we removeFirst at line 164

Not sure if it's part of a standard  algorithm, if so it would be good to paste a link to that reference, otherwise this is very cryptic.

Reply:
Agreed. Added documentation along with the link to where I got the inspiration from.